### PR TITLE
Switch to referring to ESCs by number

### LINKED ISSUE #31: Switch ESC drop-downs from City name to Region number
Available in the REGION attribute

### PR DESCRIPTION
**Pull request by eldang**
Resolves #31

Please test this one before merging!  As you'll see from the change list and commits, it wasn't quite as simple a switch as I'd expected, which has me worried I may still have missed something.

### CODE REVIEW

File: scripts/functions.js [hunk 1]
```diff
@@ -195,7 +195,7 @@ function populateZoomControl(
 	for (let i in polygons) {
 		select.append(
 			new Option(
-				polygons[i].name,
+				polygons[i].name.replace(/^0+/, ''),
```
Comment (coregis) [inline scripts/functions.js:198]: Cool, thank you for trimming the leading 0s!

**Comment by coregis** [comment]
> Resolves #31
> 
> Please test this one before merging! As you'll see from the change list and commits, it wasn't quite as simple a switch as I'd expected, which has me worried I may still have missed something.

This is working well! However I think there are a couple things that need attention before we call it good:

It looks like the drop-down for districts is bottoming out at Baird ISD; is it possible to retain the complete list so if the user wants to scroll (for days) they still can? 

image

It looks like the Education Service Centers dropdown is not getting the Work Sans styling from CSS:

image

**Comment by nein09** [comment]
> It looks like the drop-down for districts is bottoming out at Baird ISD; is it possible to retain the complete list so if the user wants to scroll (for days) they still can?

Yes! That's a TomSelect configuration option, apparently. Give it a try now.

**Comment by nein09** [comment]
ugh, I just noticed that the 'view whole state' button doesn't reset the school district select and I don't have time to fix it today.

**Comment by eldang** [comment]
I'll take a look at the CSS part later today.  There's probably either a class I should have added to the ESCs dropdown, or a CSS selector I need to add its id to.

**Comment by eldang** [comment]
I think I've sorted out both the styling and the resetting of the ISDs dropdown.

**Comment by eldang** [comment]
> You could also try `setValue('-108,25,-88,37,0', true);` if you want it to reset to the default value of "School District".

Is there a meaningful difference in behaviour?

**Comment by nein09** [comment]
> > You could also try `setValue('-108,25,-88,37,0', true);` if you want it to reset to the default value of "School District".
> 
> Is there a meaningful difference in behaviour?

It explicitly selects 'School Districts' as an option instead of clearing the selection, so it looks more like the starting state of the page because the control isn't greyed out. Aside from that, no, there is not.

**Comment by eldang** [comment]
> It explicitly selects 'School Districts' as an option instead of clearing the selection, so it looks more like the starting state of the page because the control isn't greyed out. Aside from that, no, there is not.

Ah, thank you, I'd missed that detail.  My gut feeling is that it's actually worth leaving this as the `clear()` method to not have an additional place where a zoom target is hand coded.  But it's really @coregis's call - what do you think?

**Comment by coregis** [comment]
> > It explicitly selects 'School Districts' as an option instead of clearing the selection, so it looks more like the starting state of the page because the control isn't greyed out. Aside from that, no, there is not.
> 
> Ah, thank you, I'd missed that detail. My gut feeling is that it's actually worth leaving this as the `clear()` method to not have an additional place where a zoom target is hand coded. But it's really @coregis's call - what do you think?

I agree with leaving this as the 'clear()' method, it is working well!

File: scripts/functions.js [hunk 2]
```diff
@@ -164,7 +164,7 @@ function runWhenLoadComplete() {
 			"isd",
 		);
 
-		new TomSelect("#school-districts-control", { hidePlaceholder: true });
+		new TomSelect("#school-districts-control", { hidePlaceholder: true, maxOptions: null });
```
Comment (coregis) [inline scripts/functions.js:167]: tom_selleck
Maximum Magnum

**Review by coregis — APPROVED** [review]
All of these changes look good to me, thank you both for all of your work on this! Going to merge this.

File: scripts/functions.js (hunk 2 above)
Comment (nein09) [reply scripts/functions.js:167]: LMAO

**Pull request merged**